### ISSUE TITLE
Add "short name" display to Thumbs preferences

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#184]
@hcferguson pointed out that when image names are very long, thumbnails are displayed such that you only see one thumbnail per row (instead of the usual two or three for the same GUI dimension). The problem is STScI often encounters very long image names, especially for HLA and JWST.

So, I am wondering if it is possible to add a preference to `Thumbs` plugin so it only displays a short version of image name that is configurable by the user. Possible options, where the display length and display pattern are set by user:

1. First ``n`` characters, followed by ``...`` and the extension name.
2. ``...``, followed by the last ``n`` characters and the extension name.

Currently:
```
very_very_very_long_name[ext]
    +-------------+
    | thumbnail   |
    |   image     |
    +-------------+
```

Desired display (option 1):
```
    very_ve...[ext]
    +-------------+
    | thumbnail   |
    |   image     |
    +-------------+
```

Desired display (option 2):
```
    ...g__name[ext]
    +-------------+
    | thumbnail   |
    |   image     |
    +-------------+
```

This is just an idea. Perhaps @hcferguson can provide more details or correct me if he disagrees, since he is the one who reported this.

On the other hand, maybe we can instead utilize the full control on `AstroImage` name in `Datasrc`, as you mentioned in the last comment in #175. But, automatically determining an unique short name for an arbitrary image file, or how/where we let user to customize it would need more thoughts.

Or we can close this if you think the implementation would be too complicated.

**Comment by hcferguson** [comment]
I think maybe a mouse hover showing the name might be a better solution, if that's feasible.

**Comment by pllim** [comment]
@hcferguson, that is actually easier than what I had in mind, which is a good thing! Maybe you could give my PR in #185 a try and see if that is what you wanted?

**Comment by ejeschke** [comment]
I was already thinking about option 1 combined with the mouseover, so maybe that is a good sign.

option 1 should not be hard, just need to save the full name in the `Bunch` that is created to store the thumb info and modify the label that is set, based on some name-mangling method.  Might be good to put that mangling method in a util file so it can be easily shared by `Contents` as well (although maybe `Contents`, with its resizable name column, doesn't need this modification).

Mouseover support means modifying the tooltip to add the name as the first item.

**Comment by pllim** [comment]
@ejeschke, I have an idea about the mouseover part in #185

**Comment by pllim** [comment]
Fixed by #185.
